realm/realm-library/src/objectServer/java/io/realm/mongodb/log/obfuscator/RegexPatternObfuscatorFactory.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.mongodb.log.obfuscator;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import io.realm.mongodb.Credentials;
+
+/**
+ * The RegexPatternObfuscatorFactory provides the {@link RegexPatternObfuscator}s needed to
+ * obfuscate HTTP requests being logged for a particular feature.
+ */
+public class RegexPatternObfuscatorFactory {
```

Comment:
Not 100% sure why this class is public? Despite it's name it cannot really construct anything since only the LOGIN feature is allowed, so what would end users use this for? Couldn't this just be implemented as static information in AppConfiguration?